src/pricecypher/contracts/handlers/train_models_handler.py

```diff
@@ -0,0 +1,47 @@
+from abc import abstractmethod
+from typing import Any
+
+import pandas as pd
+
+from .base_handler import BaseHandler
+from pricecypher.enums import AccessTokenGrantType
+from pricecypher.dataclasses import Models
+
+
+class TrainModelsHandler(BaseHandler):
+    """The abstract TrainModelHandler class provides a base for storing one or more models trained on a pandas
+    DataFrame.
+    Extend this class and override the `process()` method when you want to receive a DataFrame and train one or more
+    models on it.
+    The input DataFrame should be available as a pickle at the `path_in` location.
+    The output dict will be stored as a pickle at the `path_models_out` location.
+    For consistency the (unaltered) input DataFrame is stored at the `path_out` location as well.
+    """
+
+    def get_allowed_access_token_grant_types(self) -> set[AccessTokenGrantType]:
+        return set()
+
+    def get_config_dependencies(self) -> dict[str, list[str]]:
+        return dict()
+
+    def handle(self, user_input: dict[str, Any]) -> any:
+        """Handle the given `user_input`.
+        Needs a pandas DataFrame stored as a pickle at the `path_in` location. The output dict will be stored at the
+        `path_models_out` location as a pickle. For consistency the (unaltered) input DataFrame is stored at `path_out`
+        as well.
+
+        :param user_input: requires `path_in`, `path_out` and `path_models_out`.
+        :return: the remote storage path of the DataFrame.
+        """
+        input_df = self._file_storage.read_df(user_input.get('path_in'))
+        self._file_storage.write_models(user_input.get('path_models_out'), self.process(input_df))
+        return self._file_storage.write_df(user_input.get('path_out'), input_df)
+
+    @abstractmethod
+    def process(self, df: pd.DataFrame) -> Models:
```

Comment:
We can use this handler to create and train models.
Those models are the output of this handler, and so can be used by later step.
The DataFrame will be passed transparently.